backtrace-library/src/androidTest/java/backtraceio/library/BacktraceClientMetricsTest.java

```diff
@@ -0,0 +1,938 @@
+    public void setUp() {
+        context = InstrumentationRegistry.getContext();
+        credentials = new BacktraceCredentials("https://example-endpoint.com/", "");
+        BacktraceDatabase database = new BacktraceDatabase(context, context.getFilesDir().getAbsolutePath());
+
+        backtraceClient = new BacktraceClient(context, credentials, database);
+
+        BacktraceLogger.setLevel(LogLevel.DEBUG);
+    }
+
+    public class MockRequestHandler implements EventsRequestHandler {
+        public int numAttempts = 0;
+        public int statusCode = 200;
+        public String lastEventPayloadJson;
+
+        @Override
+        public EventsResult onRequest(EventsPayload data) {
+            String eventsPayloadJson = BacktraceSerializeHelper.toJson(data);
+            lastEventPayloadJson = eventsPayloadJson;
+            numAttempts++;
+
+            BacktraceResultStatus status;
+            if (statusCode == 200) {
+                status = BacktraceResultStatus.Ok;
+            } else {
+                status = BacktraceResultStatus.ServerError;
+            }
+            return new EventsResult(data, eventsPayloadJson, status, statusCode);
+        }
+    }
+
+    @Test
+    public void disableAutoSend() {
+        backtraceClient.metrics.enable(new BacktraceMetricsSettings(universeName, token, defaultBaseUrl, 0));
+        MockRequestHandler mockRequestHandler = new MockRequestHandler();
+        backtraceClient.metrics.setUniqueEventsRequestHandler(mockRequestHandler);
+
+        backtraceClient.metrics.addUniqueEvent(uniqueAttributeName[0]);
+        try {
+            sleep(100);
+        } catch (Exception e) {
+            fail(e.toString());
+        }
+        assertEquals(0, mockRequestHandler.numAttempts);
+    }
+
+    @Test
+    public void uploadSummedEventsManual() {
+        final Waiter waiter = new Waiter();
+
+        backtraceClient.metrics.enable(new BacktraceMetricsSettings(universeName, token, defaultBaseUrl, 0));
+        MockRequestHandler mockRequestHandler = new MockRequestHandler();
+        backtraceClient.metrics.setSummedEventsRequestHandler(mockRequestHandler);
+
+        backtraceClient.setSummedEventsOnServerResponse(new EventsOnServerResponseEventListener() {
+            @Override
+            public void onEvent(EventsResult result) {
+                assertEquals(0, result.getEventsPayload().getDroppedEvents());
+                String eventsPayloadJson = BacktraceSerializeHelper.toJson(result.getEventsPayload());
+                assertNotEquals(0, eventsPayloadJson.length());
+                assertEquals(BacktraceResultStatus.Ok, result.status);
+                waiter.resume();
+            }
+        });
+
+        backtraceClient.metrics.addSummedEvent(summedEventName);
+
+        assertEquals(1, backtraceClient.metrics.getSummedEvents().size());
+        // We will always have startup unique event GUID
+        assertEquals(1, backtraceClient.metrics.getUniqueEvents().size());
```

Comment:
since this is a test that check summed events - does it make sense to check unique event size here? I would recommend to move it to another test if we don't test it.

Reply:
@konraddysput this is a summed event test but in all metrics tests we will have the default unique event as well. Maybe we don't need to check it every time. If so, I can disable this check for summed events tests.

Reply:
I usually hear this rule in unit test project - test should do specific test and shouldn't test everything. If you're testing more than single thing, you should divide it into multiple tests. 

Now let's think what is important for us - in my opinion: 
- I think we should definetly have a separated test for it,
- I think it's not necessary to have this code in our unit tests code if we already have (or will have) separated unit test for it. 
- we don't need to spend additional work on refactoring everything. I think the most important thing that we will have to do is to make sure we have a separated unit test for unique event and summed event size on send. If we have it it's fine.

Reply:
Done, thank you!